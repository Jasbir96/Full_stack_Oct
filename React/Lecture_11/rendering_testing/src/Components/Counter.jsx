import React, { useState } from 'react'

function Counter() {
    const [count, setCount] = useState(0);
    const incCount = () => {
        setCount((count) => count + 1)
    }
    const decCount = () => {
        setCount((count) => count - 1)
    }
    return (
        <>
            <h1>Counter</h1>
            <button onClick={incCount}>+</button>
            <h2>Count is {count}</h2>
            <button onClick={decCount}>-</button>
     
        </>
    )
}

export default Counter;