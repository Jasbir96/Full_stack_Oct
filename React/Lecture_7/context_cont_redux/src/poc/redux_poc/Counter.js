import React from 'react'

function Counter() {
    //state management part
    const [count, setCount] = useState(0);
    const handleIncrement = () => {
        setCount(count + 1);
    }
    const handleDecrement = () => {
        setCount(count - 1);
    }
    /*****************/ 
    return (
        <>
            <button onClick={handleIncrement}>+</button>
            <h3>{count}</h3>
            <button onClick={handleDecrement}>-</button>
        </>
    )
}

export default Counter;
