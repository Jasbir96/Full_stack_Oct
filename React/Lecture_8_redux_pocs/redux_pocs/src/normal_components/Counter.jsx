import React, { useState } from 'react'

function Counter() {
    const [count, setCount] = useState(0);
    const handleIncrement = () => {
        setCount(count + 1);
    }
    const handleDecrement = () => {
        setCount(count - 1);
    }
    return (
        <div >
            <button onClick={handleIncrement}>+</button>
            <h3>{count}</h3>
            <button onClick={handleDecrement}>-</button>
        </div>
    )
}

export default Counter;


