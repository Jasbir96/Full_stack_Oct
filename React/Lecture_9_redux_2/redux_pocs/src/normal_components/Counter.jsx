import React, { useState } from 'react'

function Counter() {
    // 3.a state management
    const [count, setCount] = useState(0);
    
    // 2. bussiness logic 
    const handleIncrement = () => {
        console.log("inc");
        // 3.b
        setCount(count + 1);
    }
    
    const handleDecrement = () => {
        console.log("dec");
        //3.c
        setCount(count - 1);
    }
    return (
        // UI -> JSX
        <div >
            <button onClick={handleIncrement}>+</button>
            <h3>{count}</h3>
            <button onClick={handleDecrement}>-</button>
        </div>
    )
}

export default Counter;



