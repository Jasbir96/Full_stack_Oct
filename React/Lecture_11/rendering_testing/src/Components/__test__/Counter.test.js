/***
 * count=0
 * inceremnt 
 * decrement
 * */
import { render, screen, fireEvent } from "@testing-library/react";
import Counter from "../Counter";
describe("counter test cases", () => {

    test("snapshot for counter", () => {
        const { asFragment } = render(<Counter></Counter>);
        expect(asFragment()).toMatchSnapshot();
    })

    test("inital state check", () => {
        // rendering it in isolation 
        render(<Counter></Counter>);
        //selection
        const countText = screen.getByText("Count is 0");
        const plusText = screen.getByText("+");
        const minusText = screen.getByText("-");
        // we are verifying
        expect(countText).toBeInTheDocument();
        expect(plusText).toBeInTheDocument();
        expect(minusText).toBeInTheDocument();
    })
    test("inc by 1", () => {
        // rendering it in isolation 
        render(<Counter></Counter>);
        const plusText = screen.getByText("+");
        fireEvent.click(plusText);
        fireEvent.click(plusText);
        const isOnePresent = screen.getByText("Count is 2");
        expect(isOnePresent).toBeInTheDocument();
        //selection
        // const countText = screen.getByText("Count is 0");
        // const plusText = screen.getByText("+");
        // const minusText = screen.getByText("-");
        // // we are verifying
        // expect(countText).toBeInTheDocument();
        // expect(plusText).toBeInTheDocument();
        // expect(minusText).toBeInTheDocument();
    })
    test("dec by 1", () => {
        // rendering it in isolation 
        render(<Counter></Counter>);
        const minusText = screen.getByText("-");
        fireEvent.click(minusText);
        fireEvent.click(minusText);
        const isOnePresent = screen.getByText("Count is -2");
        expect(isOnePresent).toBeInTheDocument();
        //selection
        // const countText = screen.getByText("Count is 0");
        // const plusText = screen.getByText("+");
        // const minusText = screen.getByText("-");
        // // we are verifying
        // expect(countText).toBeInTheDocument();
        // expect(plusText).toBeInTheDocument();
        // expect(minusText).toBeInTheDocument();
    })
})
