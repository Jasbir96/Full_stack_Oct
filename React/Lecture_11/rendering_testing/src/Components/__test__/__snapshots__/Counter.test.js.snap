// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter test cases snapshot for counter 1`] = `
<DocumentFragment>
  <h1>
    Counter
  </h1>
  <button>
    +
  </button>
  <h2>
    Count is 0
  </h2>
  <button>
    -
  </button>
</DocumentFragment>
`;
